sympy/core/power.py

```diff
@@ -22,9 +22,21 @@
 
 def isqrt(n):
     """Return the largest integer less than or equal to sqrt(n)."""
-    if n < 17984395633462800708566937239552:
-        return int(_sqrt(n))
-    return integer_nthroot(int(n), 2)[0]
+    if n < 0:
+        raise ValueError("n must be nonnegative")
```

Comment:
I believe that this check can be removed. Both `_sqrt` and `integer_nthroot` already have checks for non-negative n. If still the check is necessary, then can you please share the reason so that I can keep that it in mind for future reviews of other PRs.

Reply:
It's needed for values of `n` between `-1` and `0`. I added tests for those cases, so (if I got the tests right) you _should_ find that removing this check causes the tests to fail.

Reply:
Though if we restrict to integral `n` as @oscarbenjamin suggested, then yes, the non-negativity check is redundant.

Reply:
> I wonder now whether we should just use mpmath's sqrtrem and let mpmath use math.isqrt or qmpy as appropriate...

I think that would make a lot of sense.